src/Standards/Generic/Sniffs/PHP/SyntaxSniff.php

```diff
@@ -48,20 +41,8 @@ public function register()
      */
     public function process(File $phpcsFile, $stackPtr)
     {
-        if ($this->phpPath === null) {
-            $this->phpPath = Config::getExecutablePath('php');
-            if ($this->phpPath === null) {
-                // PHP_BINARY is available in PHP 5.4+.
-                if (defined('PHP_BINARY') === true) {
-                    $this->phpPath = PHP_BINARY;
-                } else {
-                    return;
-                }
-            }
-        }
-
         $fileName = escapeshellarg($phpcsFile->getFilename());
-        $cmd      = escapeshellcmd($this->phpPath)." -l -d display_errors=1 -d error_prepend_string='' $fileName 2>&1";
+        $cmd      = escapeshellcmd(Config::getExecutablePath('php'))." -l -d display_errors=1 -d error_prepend_string='' $fileName 2>&1";
```

Comment:
As the same sniff object is used for every file, this change results in calling Config::getExecutablePath('php') for every single file being checked. You removed all the path caching that this sniff had in place to improve performance, so you'll need to put the private member var back, along with the IF statement above. It can just be made far simpler now.

Reply:
My motivation for not putting it back, is that the `Config` class already caches it. Its kinda like that class's responsibility to cache config values, right?

Reply:
Config:: getExecutablePath() will call Config::getConfigData() which will call Config::getAllConfigData() and an isset(). This will happen for every file being checked.

It doesn't sound like a lot, but it has an impact at scale. When I get time to do profiling, these are the sort of caches I put in to try and cut down the number of overall function calls and lines of code executed. I'd prefer the cache to be left in place to ensure a consistent performance profile across versions.